src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -154,6 +159,19 @@ private CatalogObjectRevisionEntity buildCatalogObjectRevisionEntity(String kind
         return buildMetadataWithLink(bucketId, result);
     }
 
+    public byte[] getCatalogObjectsAsZipArchive(Long bucketId, List<String> catalogObjectsNames) {
+        List<CatalogObjectRevisionEntity> revisions = new ArrayList<>();
+
+        for (String name : catalogObjectsNames) {
+            CatalogObjectRevisionEntity catalogObjectRevision = catalogObjectRevisionRepository.findDefaultCatalogObjectByNameInBucket(bucketId,
+                                                                                                                                       name);
+            if (catalogObjectRevision == null)
```

Comment:
Why returning null when an object is missing ? 
If a missing object is not a normal behavior it would be better to through an exception. Otherwise if it is not an issue I think that returning the a zip without the missing object would be a better solution.